src/main/java/org/spongepowered/api/item/inventory/InventoryArchetypes.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.item.inventory;
+
+import com.flowpowered.math.vector.Vector2i;
+import org.spongepowered.api.item.inventory.property.AcceptsItems;
+import org.spongepowered.api.item.inventory.property.InventorySize;
+import org.spongepowered.api.item.inventory.property.SlotIndex;
+import org.spongepowered.api.item.inventory.property.TitleProperty;
+
+/**
+ * An enumeration of {@link InventoryArchetype}s.
+ */
+public class InventoryArchetypes {
+
+    // TODO Dummies
+
+    // Base Archetypes
+
+    // TODO how to start?
```

Comment:
This is an interesting question.

@Mumfrey , considering the API...can you help @Faithcaio figure out the correct way to kick off implementation? I'll give a scenario that stumps me as well.

```
InventoryArchetype COMBINED = InventoryArchetype.builder()
    .with(InventoryArchetypes.CHEST)
    .with(InventoryArchetypes.DISPENSER)
    .with(InventoryArchetypes.FURNACE)
    .property(TitleProperty.of(new FixedTranslation("Lulzy Inventory"))
    .build("combined", "Combined");

Inventory myLulzyInventory = Inventory.builder().of(COMBINED).build();

player.openInventory(myLulzyInventory);
```

Considering the above code, I've no idea what Container to show in impl. How is that defined? Is the Container a Property of the Inventory? If you use a Container that doesn't match the Vanilla archetype in question, is that a fail fast?